testdata/p4_14_samples_outputs/issue1237-first.p4

```diff
@@ -0,0 +1,81 @@
+#include <core.p4>
+#include <v1model.p4>
+
+@name("backs") header backs_0 {
+    bit<16> sharps;
+    bit<16> peptides;
+    bit<16> whirs;
+    bit<16> breasted;
+    bit<16> browning;
+    bit<32> jews;
+}
+
+@name("affairs") header affairs_0 {
+    bit<48> flaccidly;
+    bit<16> sleeting;
+}
+
+struct metadata {
+}
+
+struct headers {
+    @name(".expressivenesss") 
+    backs_0   expressivenesss;
+    @name(".kilometer") 
+    affairs_0 kilometer;
+}
+
+parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
+    @name(".start") state start {
+        packet.extract<affairs_0>(hdr.kilometer);
+        packet.extract<backs_0>(hdr.expressivenesss);
+        transition accept;
+    }
+}
+
+control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
+    @name(".mooneys") action mooneys() {
+        hdr.expressivenesss.breasted = hdr.expressivenesss.breasted - hdr.expressivenesss.peptides;
+        hdr.kilometer.sleeting = hdr.kilometer.sleeting + 16w65529;
+    }
+    @name(".conceptualization") table conceptualization {
+        actions = {
+            mooneys();
+            @defaultonly NoAction();
+        }
+        key = {
+            hdr.kilometer.isValid()      : exact @name("kilometer.$valid$") ;
+            48w0                         : lpm @name("0") ;
```

Comment:
It's unclear to me if the control plane API for this argument would be generated correctly. Would the argument be named '0'?

Reply:
That's a good question. Today this is a warning, maybe it should be an error?
@antoninbas : what should happen when a key field is a constant expression?

Reply:
I have a hard time imagining a situation where someone intentionally wants a compile-time constant value as a table key expression.  It seems that a correct implementation could leave that key in, and always search with that field equal to the constant value (0 in this example), but if the control plane ever installed a table entry that did not match 0 for that field, such a table entry would be impossible to match.

It seems like it justifies at least a warning from the compiler, if not an error.  If it is a warning, it isn't clear that much effort should be made in later tools to support this case, unless someone can think of a valid use case for it, so perhaps an error is preferable?

Reply:
I think one acceptable solution that is probably also the easiest to implement is to print a warning in the p4RuntimeSerializer and omit the field altogether from the P4Info message. This way we do not put any burden on the P4Runtime implementers, for something that doesn't seem to have any actual use-cases.

Reply:
Since we already have a warning I declare this PR ready for review.